pkg/memtier/policy_age.go

```diff
@@ -216,21 +236,10 @@ func (p *PolicyAge) Dump(args []string) string {
 				time.Duration(p.config.SwapOutMs)*time.Millisecond,
 				time.Duration(0))
 		} else {
-			for _, timeSpec := range strings.Split(strings.TrimSpace(args[1]), ",") {
-				timeDur, err := parseTimeDuration(timeSpec)
-				if err != nil {
-					return fmt.Sprintf("invalid TIMESPEC %q: %s", timeSpec, err)
-				}
-				timeDurations = append(timeDurations, timeDur)
-			}
-			if len(args) == 3 {
-				for _, pidSpec := range strings.Split(strings.TrimSpace(args[2]), ",") {
-					pid, err := strconv.Atoi(pidSpec)
-					if err != nil {
-						return fmt.Sprintf("invalid pid: %q", pidSpec)
-					}
-					pids = append(pids, pid)
-				}
+			var errInfo string
+			errInfo, timeDurations, pids = p.parseAndUpdateTimeAndPIDs(args, timeDurations, pids)
```

Comment:
Would you split the parse function into two: one parsing time durations, and the other parsing pids? Both functions should return an error as the last return value as usual in Go. And both of them should be given the string to be parsed directly, so that the functions do not need to know from which indexes in a slice they should find a string to be parsed.

With these functions the code here would look like something like this:
```
parsedDurations, err := parseTimeDurations(args[1])
if err != nil {
    return err.Error()
}
timeDurations = append(timeDurations, parsedDurations...)

if len(args) == 3 {
    parsedPids, err := parsePids(args[2])
    if err != nil {
        return err.Error()
    }
    append(pids, parsedPids...)
}
```

This will not "reduce complexity" in static analysis as much because there will be more and nested ifs. However, it reduces more complexity in the way is unfortunately not calculated by the static analysis tool. Every function has only one purpose, return values follow go practices everywhere in the codebase, and assumptions on the order of elements in the args slice do not spread outside the dump function.